### PR TITLE
Feat: custom project and organization TTL policy

### LINKED ISSUE #572: Custom Project and Organization TTL Policy
#### Why
Users want to be able to customize the Max and Default TTL for environment  and not to be limited to the current presets

#### How
For [Organization Policy](https://registry.terraform.io/providers/env0/env0/latest/docs/resources/organization_policy) and [Project Policy](https://registry.terraform.io/providers/env0/env0/latest/docs/resources/project_policy) we should change `max_ttl` and `default_ttl` to be any string of type `<number>-<M/w/d/h>` (still omit for infinity) and it should not be limited to the presets.
Those are only user facing changes, we don't want to make any api changes, BE is ok with those kind of strings.
So I think the only requires change is here is to remove the limitation on the allowed strings and just force validation on the `<number>-<M/w/d/h>` schema.

### PR DESCRIPTION
**Pull request by TomerHeber**
### Issue & Steps to Reproduce / Feature Request
resolves #572 

### Solution

1. Updated both resources `env0_organization_policy` and `env0_project_policy` to support custom TTL values.
2. updated acceptance tests.
3. added conversion functions and unit tests.
4. updated integration tests.

### CODE REVIEW

File: env0/utils.go [hunk 1]
```diff
@@ -453,3 +457,34 @@ func writeResourceDataEx(prefix string, i interface{}, d *schema.ResourceData) e
...
+	number, err := strconv.Atoi(numberStr)
+	if err != nil {
+		return 0, fmt.Errorf("invalid TTL format %s - not a number: %w", ttl, err)
+	}
+	// M/w/d/h
+
+	var hours int = number
+
+	switch rangeType := match[2]; rangeType {
+	case "M":
```
Comment (TomerHeber) [inline env0/utils.go:481]: Assumed that 1 Month equals 30 days.

File: env0/utils.go [hunk 2]
```diff
@@ -453,3 +457,34 @@ func writeResourceDataEx(prefix string, i interface{}, d *schema.ResourceData) e
 	}
 	return writeResourceDataSlice([]interface{}{i}, prefix, d)
 }
+
+func ttlToDuration(ttl string) (time.Duration, error) {
+	if ttl == "Infinite" || ttl == "inherit" {
```
Comment (TomerHeber) [inline env0/utils.go:462]: Inherit is hard to predict.
Worst case scenario, the backend call will fail.

File: env0/resource_project_policy.go [hunk 3]
```diff
@@ -152,10 +158,18 @@ func resourcePolicyUpdate(ctx context.Context, d *schema.ResourceData, meta inte
...
+		return diag.Errorf("invalid default ttl: %v", err)
+	}
+
+	maxTtl, err := getPolicyTtl(payload.MaxTtl)
+	if err != nil {
+		return diag.Errorf("invalid max ttl: %v", err)
+	}
+
+	if maxTtl < defaultTtl {
+		return diag.Errorf("default ttl must not be larger than max ttl: %d %d", defaultTtl, maxTtl)
```
Comment (eladmosh) [inline env0/resource_project_policy.go:172]: anyway to avoid the code duplication here and in organization?

File: env0/utils.go (hunk 2 above)
Comment (eladmosh) [inline env0/utils.go:462]: In case of `infinite` user will omit the ttl, so it will be `nil` and not `Infinite`

File: env0/utils.go (hunk 1 above)
Comment (eladmosh) [reply env0/utils.go:481]: Isn't there a way to calc a "real" month using https://pkg.go.dev/time?
This will create problems

Comment (TomerHeber) [reply env0/utils.go:481]: 1-M would mean something different every month.
In other words, 1-M isn't well-defined when it's configured.

File: env0/resource_project_policy.go (hunk 3 above)
Comment (TomerHeber) [reply env0/resource_project_policy.go:172]: I think there is. I'll check...

File: env0/utils.go (hunk 1 above)
Comment (eladmosh) [reply env0/utils.go:481]: nvm, just keep it

File: env0/utils.go (hunk 2 above)
Comment (TomerHeber) [reply env0/utils.go]: Generally not an issue, but made some refactoring.
d12f08e06636e169d9c306a715ceb5f687e37d60

File: env0/resource_project_policy.go (hunk 3 above)
Comment (TomerHeber) [reply env0/resource_project_policy.go]: Refactored and removed code duplication.
d12f08e06636e169d9c306a715ceb5f687e37d60

**Review by yaronya — APPROVED** [review]

**Pull request merged**